### PR TITLE
depends: only use explicit CC for mingw32 in intel-ipsec-mb

### PR DESCRIPTION
**Pull request by patricklodder**
The `CC` override in `intel-ipsec-mb.mk` fails for linux builds when done in a docker container due to it inserting "pc" into the host variable when not otherwise specified.

This fixes it by only overriding if we're building with mingw.

Note 1: this is the laziest solution I could come up with - ideally we'd figure out why intel-ipsec-mb doesn't grab the correct CC - I'd yield this PR to one that does that instead in case someone is interested in figuring that out.
Note 2: I'll mark this as do-not-merge and blocked until #3136 is in.

### CODE REVIEW

**Comment by patricklodder** [comment]
Rebased on top of 4c31ab1240, this is now ready for review.

**Comment by chromatic** [comment]
Do you have a recommended Docker image to test this? I'm not sure if I'm looking for reproduction instructions; seems like this is better to test if I come up with some of them myself.

**Comment by patricklodder** [comment]
To reproduce the issue: 

```Dockerfile
FROM ubuntu:22.04
WORKDIR /build

# depends systems prereqs
RUN apt update \ 
 && apt install -y autoconf automake binutils-gold ca-certificates curl \
                   build-essential faketime git libtool pkg-config python3 bison

RUN git clone https://github.com/dogecoin/dogecoin.git \
 && cd dogecoin \
 && git checkout 4c31ab1 \
 && make -C depends AVX2=1 native_nasm intel-ipsec-mb
```

Will fail with:

```
make[2]: x86_64-pc-linux-gnu-gcc: No such file or directory
make[2]: *** [Makefile:768: obj/chacha20_poly1305.o] Error 127
```

**Comment by edtubbs** [comment]
ACK, tested change with docker build

**Review by edtubbs — APPROVED** [review]

**Review by chromatic — APPROVED** [review]
ACK with the Dockerfile.

I reproduced the error, then modified the Dockerfile to try this commit and the build succeeded.

**Comment by xanimo** [comment]
I would like to double check something and would appreciate a little bit more time if possible before merging.

**Comment by patricklodder** [comment]
Added do-not-merge-yet to see what comes out of triage with #3162

**Comment by xanimo** [comment]
I've just reproduced the error and performed a couple more tests. Ideally I'd spend a bit more time on this but I'm short on time and don't want to hold this up. 

Regardless I'm of the opinion the `CC` override is unnecessary and we should strike it altogether as I was able to build depends for our default host triplet `x86_64-pc-linux-gnu` and without explicitly setting the `HOST` variable to `x86_64-linux-gnu` which was the previous work around I used as implemented in CI for experimental linux when triaging #3162. I think this would be the best route as I can't imagine a scenario of building `x86_64-w64-mingw32` without explicitly setting the `HOST` variable either.

**Comment by patricklodder** [comment]
> Regardless I'm of the opinion the CC override is unnecessary

So are you saying you have a solution where the entire override can be removed? Because I stated all the way up that I'd yield this PR to one that achieved that, so then we can just close this.

**Comment by xanimo** [comment]
Oh yeah, I should've clarified my comment wasn't meant to block. I had just run a couple tests after reproducing the error you gave instructions for which entailed removing that line in the intel-ipsec-mb.mk file which allowed me to run `make -C depends AVX2=1 ...` without setting the `HOST` variable for `x86_64-pc-linux-gnu`. That aside I forgot to test your solution commit but utACK nonetheless.

**Comment by patricklodder** [comment]
Removed do-not-merge because of @xanimo's utACK [here](https://github.com/dogecoin/dogecoin/pull/3146#issuecomment-1309259728)

**Review by xanimo — APPROVED** [review]
ACK. Tested on focal for x86_64-pc-linux-gnu and x86_64-w64-mingw32.

**Comment by abosaleh82** [comment]
hello sir,

i don't understand what you mean>>>
can you explain me more please

**Comment by chromatic** [comment]
Given the removed do-not-merge and no changes since my tested ACK, I'm going to merge this in 24 hours, as that's the spirit of our policy if not the letter.

**Pull request merged**
